tests/2_functional-tests.js

```diff
@@ -18,19 +10,36 @@ suite('Functional Tests', function() {
   suite('API ROUTING FOR /api/threads/:board', function() {
     
     suite('POST', function() {
-      
+       test('create 2 new threads (one is deleted in the delete test)', function(done) {
```

Comment:
@nhcarrigan I think we may have a different issue.  We are depending on the user reading the text of the test to know what to write for this particular test. Ideally, the `/learn` page would explain what tests need to be written and not depend on the user to be able to read the text of the test in the boilerplate.  Why, because all instructions or instruction-like parts of a project need to be translated and we can not translate the boilerplate code.

Reply:
Ah, I see. 

Okay - I can run a refactor on this (and the other) to:
* Add number comments to identify the tests, as in the mocha chai boilerplate
* Move the corresponding instructions to a numeric list in the `/learn` instructions

I think this will need to be done for all of the QA and InfoSec projects, though, as they all follow this format.

Reply:
> I think this will need to be done for all of the QA and InfoSec projects, though, as they all follow this format.

Yes, I agree.  @moT01 Maybe we have this done in a separate PR for each QA and InfoSec project, so we can at least merge these "as-is"?  We will just not close freeCodeCamp/freeCodeCamp#39721 until this has been done for all these projects? Plus the scope of those new PRs will be very specific.  Or, should we them go ahead and fix each of the existing open PRs?

Reply:
Tough to say, many of these have certainly gone outside the scope of the defined issue, but I'm fine with that since changing things later will result in another round of PR's and reviews for each boilerplate with the issue. We could look at them case by case - this particular PR and it's related learn PR came in yesterday and are just starting the review process, so I wouldn't mind changing these before they're merged. Others that have been reviewed thoroughly and are ready to be merged, it may not make sense to now change those up further. I suppose it also depends if @nhcarrigan wants to make the changes. If he's willing - which it seems he is - we could try and change them - I don't mind saving it for another issue/PR either.

To try and clarify, what would you do here?
```diff
-  test('create 2 new threads (one is deleted in the delete test)', function(done) {
+  test('#1', function(done) {
```
Number all the tests like that, and then put in the instructions on learn a list of the test text and the number they go with? I know we've talked about that a little bit, but maybe we should see what others think about it before we make a decision to change them all. If we really want to move all the instructions to learn, I don't see much else for options though. And if these don't make their way to learn somehow, they won't get translated. I do kind of feel like this could use a little discussion - maybe someone can come up with a better method or maybe people will think this is too confusing or something.

Reply:
I like this approach.  However, I think the verbiage for the tests should be in the instructions sections above and the last user story in the tests would refer the user to fulfill the tests specified in the instructions.

Reply:
> I think the verbiage for the tests should be in the instructions sections above and the last user story in the tests would refer the user to fulfill the tests specified in the instructions.

This was my thought as well - I thought of something like:
```md
## Instructions
<section id='instructions'>

1. Set `NODE_ENV` to test without quotes when ready to write tests and DB to your databases connection string (in `.env`)
2. Recommended to create controllers/handlers and handle routing in `routes/api.js`
3. You will add any security features to `server.js`

Write the following unit tests in `tests/1_unit-tests.js`:
1. Test description

Write the following functional tests in `tests/2_functional-tests.js`:
1. Test description
</section>
```

That way we could remove the "unit test" section if applicable.

Reply:
@moT01 and I were just discussing how if we go with putting the instructions for the tests to write in the `Instructions` section, then maybe we could remove all the `test('#1', function(done) {` and just leave the `suite`s to force the user to do more of the work.  The instructions would just tell them what tests to write and reference which `suite` to put them in.  What do you think?

Reply:
```js
const chaiHttp = require('chai-http');
const chai = require('chai');
const assert = chai.assert;
const server = require('../server');

chai.use(chaiHttp);

suite('Functional Tests', function() {
    
});
```

So reducing it down to this? I like that approach. The client test will confirm the user has written enough tests to match the user story requirements.
(I'd be willing to go one step further and remove the imports as well)

Reply:
> (I'd be willing to go one step further and remove the imports as well)

Which ones were you thinking?

Reply:
Bringing it down to only:
```js
suite('Functional Tests', function() {
    
});
```

It'll require more thinking from the user - though it could also create more confusion than we'd like.

Reply:
I could be ok with this.  @moT01 @scissorsneedfoodtoo What do you think?

Reply:
@RandellDawson, that looks good to me. By this point learners should be familiar with import statements. Simplifying the test files like this would also give them a chance to write tests in either BDD or TDD styles.

Reply:
Alright - I can start shipping this change to the other projects as well, then. 🙂

Reply:
Awesome! Thank you for all of your hard work on these projects @nhcarrigan, and for everything else you're helping us with.

Reply:
Always glad to help! I'll start shipping these changes tomorrow!